clients/elrond/client.go

```diff
@@ -110,9 +117,14 @@ func NewClient(args ClientArgs) (*client, error) {
 		return nil, clients.ErrNilAddressConverter
 	}
 
+	proxyWithCacher, err := blockchain.NewElrondProxyWithCache(args.Proxy, time.Second*time.Duration(args.ProxyCacherExpirationTime))
```

Comment:
I would move this from here and create the Proxy cacher instance just after the instantiation of the Elrond Proxy (factory package). In this way this caching Proxy instance can be reused elsewhere